ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -70,27 +72,44 @@ targetPeers :: (MonadSTM m, Ord peeraddr)
             => PeerSelectionActions peeraddr peerconn m
             -> PeerSelectionState peeraddr peerconn
             -> Guarded (STM m) (TimedDecision m peeraddr peerconn)
-targetPeers PeerSelectionActions{readPeerSelectionTargets}
+targetPeers PeerSelectionActions{ readPeerSelectionTargets = readTargets,
+                                  currentTargets,
+                                  churnMutex }
             st@PeerSelectionState{
               publicRootPeers,
               localRootPeers,
               targets,
-              ledgerStateJudgement,
+              ledgerStateJudgement = lsj,
               bootstrapPeersFlag,
-              hasOnlyBootstrapPeers
+              hasOnlyBootstrapPeers,
+              useGenesisFlag
             } =
     Guarded Nothing $ do
-      targets' <- readPeerSelectionTargets
-      check ( isNodeAbleToMakeProgress bootstrapPeersFlag
-                                       ledgerStateJudgement
-                                       hasOnlyBootstrapPeers
-            && targets' /= targets
-            && sanePeerSelectionTargets targets'
-            )
+      -- Possible values:
+      -- Nothing: churn governor is active
+      -- Just lsj: Previous ledger state judgement in Genesis mode, or
+      -- Just TooOld when not using Genesis
+      -- we can't use readTMVar because we will block when churn is
+      -- on thus making it useless.
+      mode <- tryReadTMVar churnMutex
+      targets' <- if useGenesisFlag && not (churning mode)
```

Comment:
I think I can improve naming, here mode is used only in the condition check to avoid a case where we could reset targets on the basis of ledger state judgement changing while churning is active.

Reply:
If we don't need the value itself only the lock state it's better to use `TMVar m ()`.